test/sql/types/union/union_test.test

```diff
@@ -0,0 +1,48 @@
+# name: test/sql/union/union_test.test
+# description: Test union type
+# group: [union]
+
+statement ok
+CREATE TABLE tbl(u UNION(i INTEGER, v VARCHAR));
```

Comment:
Some more suggestions for tests:

* Inserting multiple values into a UNION table at once (e.g. `INSERT INTO tbl SELECT i FROM range(10) tbl(i)`)
* Inserting `UNION` values directly (e.g. `CREATE TABLE tbl2 AS SELECT * FROM tbl`, or `INSERT INTO tbl2 SELECT * FROM tbl`, where `tbl` and `tbl2` have UNION columns with multiple values)
* Duplicate name in union entry
* Attempting to extract a non-existent column from a union
* Extracting unions of different vector types (e.g. add a `WHERE` clause to create a dictionary vector)
* Manual casts to union types
* Storing NULL values in union columns
* Unions with multiple identical types (e.g. `UNION(i INTEGER, j INTEGER)`)
* Lists inside unions (e.g. `UNION(i INTEGER[], j INTEGER)`)
* Lists OF unions (e.g. `UNION(i INTEGER, s VARCHAR)[]`)
* Nested unions (union inside of another union)
* `UNION` different `UNION` types (hah) - (i.e. `SELECT 1::UNION(i INTEGER) UNION ALL SELECT 2::UNION(i BIGINT)`).
* GROUP BY/DISTINCT on union columns
* Persistent storage of union types (see the tests in `test/sql/storage`)